### PR TITLE
fix: change timing for the stopEdit() call on focusOut cell editor

### LINKED ISSUE #149: Cannot select value from a custom Datepicker column
Vaadin version: 14.4.2 

Observed behavior:
* The value is updated when typing the value
* The value is updated when opening the popup and pressing key up / key down, then selecting a value using the mouse
* The value is NOT updated when selecting a value from the popup, without using the keyboard

(this is unrelated to #95 since it describes some styling issues)

### Example
```
GridPro<Item> gridpro = new GridPro<>();
DatePicker datePicker = new DatePicker();
gridpro.addEditColumn(item->item.getValue(), new LocalDateRenderer<>(item->item.getValue(), "dd/MM/yyyy")).custom(datePicker,(item,date) -> {
  item.setValue(date);
  System.out.println(date);
});
gridpro.setItems(new Item());
```

### PR DESCRIPTION
**Pull request by sosa-vaadin**
Fixes #149

### CODE REVIEW

File: test/edit-column-overlay.test.js [hunk 1]
```diff
@@ -6,10 +6,13 @@ import { createItems, enter, flushGrid, getCellEditor, getContainerCell } from '
 import '../vaadin-grid-pro.js';
 import '../vaadin-grid-pro-edit-column.js';
 
-function clickOverlay(element) {
+async function clickOverlay(element) {
   const focusout = new CustomEvent('focusout', { bubbles: true, composed: true });
   element.dispatchEvent(focusout);
 
+  // add a microTask in between
+  await new Promise((resolve) => resolve());
```
Comment (web-padawan) [inline test/edit-column-overlay.test.js:14]:
```suggestion
  await Promise.resolve();
```

File: test/edit-column-overlay.test.js [hunk 2]
```diff
@@ -69,14 +72,15 @@ const fixtures = {
...
+    it('should not stop editing when focusing input related overlay', async () => {
       enter(dateCell);
       const datePicker = getCellEditor(dateCell).querySelector('vaadin-date-picker');
       datePicker.click();
 
-      clickOverlay(datePicker);
+      await clickOverlay(datePicker);
       grid._debouncerStopEdit && grid._debouncerStopEdit.flush();
 
+      await new Promise((resolve) => requestAnimationFrame(resolve));
```
Comment (web-padawan) [inline test/edit-column-overlay.test.js:83]:
```suggestion
      await nextFrame();
```

And also update the import above to look like this:

```js
import { fixtureSync, nextFrame } from '@open-wc/testing-helpers';
```

**Review by web-padawan — CHANGES_REQUESTED** [review]

File: test/edit-column-overlay.test.js (hunk 1 above)
Comment (sosa-vaadin) [reply test/edit-column-overlay.test.js]: Done.

File: test/edit-column-overlay.test.js (hunk 2 above)
Comment (sosa-vaadin) [reply test/edit-column-overlay.test.js]: Done.

**Review by tomivirkki — APPROVED** [review]

**Review by web-padawan — APPROVED** [review]

**Pull request merged**